### PR TITLE
new DropoutAllchannels operator

### PR DESCRIPTION
**Pull request by kadeng**
Like Dropout operator, but drops / keeps entire pixels from image or image-like input records (such as the output of 2D convolution)

Retrying this as a new pull request (See #1490 ) . Last time, Travis seems to have picked up an outdated version of one file..

### CODE REVIEW

**Comment by tqchen** [comment]
@antinucleon can you review this?

File: src/operator/dropout_allchannels.cc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Operator *CreateOp<cpu>(DropoutAllchannelsParam param) {
+  return new DropoutAllchannelsOp<cpu>(param);
+}
+
+// DO_BIND_DISPATCH comes from operator_common.h
+Operator *DropoutAllchannelsProp::CreateOperator(Context ctx) const {
+  DO_BIND_DISPATCH(CreateOp, param_);
+}
+
+DMLC_REGISTER_PARAMETER(DropoutAllchannelsParam);
```
Comment (piiswrong) [inline src/operator/dropout_allchannels.cc]: capitalize Channel

File: src/operator/dropout_allchannels-inl.h [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                        const std::vector<OpReqType> &req,
+                        const std::vector<TBlob> &in_grad,
+                        const std::vector<TBlob> &aux_states) {
+    using namespace mshadow;
+    using namespace mshadow::expr;
+    CHECK_EQ(out_grad.size(), 1);
+    CHECK_EQ(in_grad.size(), 1);
+    Stream<xpu> *s = ctx.get_stream<xpu>();
+    Tensor<xpu, 4> grad = in_data[dropout_allchannels::kOut].get<xpu, 4, real_t>(s);
+    Tensor<xpu, 4> mask = in_data[dropout_allchannels::kMask].get<xpu, 4, real_t>(s);
```
Comment (piiswrong) [inline src/operator/dropout_allchannels-inl.h]: out data?

**Comment by piiswrong** [comment]
Backward doesn't seem correct. Could you add a test case?

**Comment by kadeng** [comment]
You're right, I should have checked the Backward implementation. This variant should be correct now, I also changed the spelling of DropoutAllchannels to DropoutAllChannels

**Comment by kadeng** [comment]
This pull request is probably ready to merge. Can somebody review this ? :) Thanks

**Comment by tqchen** [comment]
close this for now as it is likely can be implemented together with #2511

**Comment by horserma** [comment]
Any update on this? Has this dropallchannels operator been one of the official operators? I did not find it in the document.

**Pull request closed without merging**